### DISCUSSION TITLE
Is there any goal in the roadmap of regolith desktop to make distribution-free?

### DISCUSSION THREAD

**Discussion by abdyek** [regolith-linux/regolith-desktop#822] [Q&A]
Hello everyone,

I want to use regolith desktop on other distros. Is there any official way for it? If no, do you have it on your roadmap?

**Comment by akspecs** [comment]
having a regolith tiling desktop experience available outside of debian (ubuntu) would be great!  i recommended regolith to my partner who i pair program with.  they like the idea of a tiling wm, but never had the time to configure or learn one to it's fullest extent.  regolith bridged that gap immensely, and they are loving it!

with future wayland / sway support planned, the only downside i can think of is lack of widespread distro availability.  but, nothing is stopping you from trying to package regolith for another distro of your choosing ;)

i'm aware of https://aur.archlinux.org/packages/regolith-full which currently has v. 1.6 packaged in the AUR.  the package maintainer is working on getting v. 2+ packaged -- and i'll defnitely give that a spin once it is available (or try porting / packaging it myself ... )

i too only wish regolith's  setup + configs were more widespread outside of debian based distros.  other than that, i'm eagerly anticipating sway support :)

**Comment by kgilmer** [comment]
A primary design aspect of Regolith is to use the package manager's capabilities for system configuration.  For example, to configure widgets in the bar, you can install/remove packages that contribute various bits of configuration.  A cost to this approach is greater complexity in the packaging of the desktop environment.  Additionally, Regolith is very different than typical Linux DEs.  This makes the porting work more complex, and just more of it.  There have been some attempts in the past to port Regolith to other packaging systems such as Arch, Fedora.  I've offered to help out where I can but lack the time to be able to learn, test, and support Regolith beyond Ubuntu/Debian.  If there were users adept at packaging for other systems, I would be happy to help them get started in packaging.  Testing support and distro expertise are two areas we lack people in to move this forward IMO.

**Reply by trharmon0** [reply]
This is the primary use-case for the various configuration management tools. They are, arguably, not favored for high-deployment-frequency server-side apps with the advent of cluster managers and similar tech but they are still very effective for crafting and validating systems which are not immutable and need to be mutated over time.

Of the ones out there now, I would make the argument the forked version of Puppet(OpenVox), due to its Resource Abstraction Layer, would be effective. Someone might make the argument for Ansible as well but I don't trust Red Hat's stewardship when it comes to keeping it friendly and collaborative with the Open Source community and it lacks the concept of a RAL completely which balloons the CM code base when you start worknig with multiple OSes ... YAML. YAML are far as the eye can see!. Similar situation for Chef and Salt as for Ansible though different "owners".

I'm tagging in on this because I worked for Puppet Labs back in the day and have considerable Puppet experience (though admittedly it would need rewarmed) and, more selfishly, my employer has forced Linux users to vanilla GNOME on Fedora and so I'm suffering mightily right now. ;)

Before I would want to even attempt to Puppetize the install I would need some sort of confirmation that that effort would not be dismissed offhand. 

lmk

EDIT: To be clear, I envision Puppet being used to automate + validate the install and NOT as a permanent addition to a system which has been Regolith'ed. For those not familiar, 'puppet apply'/no-puppet-master model and Puppet removed post-install.

**Reply by trharmon0** [reply]
Anticipating a possible go-forward from leadership, I did some digging on the current state of Ansible. It does appear to have a reduced-functionality set of constructs which correlate to Puppet/OpenVox RAL so maybe its technically viable. My reservations about stewardship persist. I'm open to it as an option depending on feedback.

lmk

**Reply by trharmon0** [reply]
@kgilmer If there's any interest in this an effective first step would be to talk to an SME on the CI/CD system for Regolith and understand the build + test conventions used for the project. I have considerable on-prem CI/CD infra would could be used to do module testing without having a bunch of cloud spend.
